internal/extensions/extensions.go

```diff
@@ -9,8 +9,8 @@ import (
 	"github.com/canonical/lxd/shared"
 )
 
-var internalExtensionRegex = regexp.MustCompile(`^internal:([a-z0-9]|[a-z0-9]_[a-z0-9])+$`)
-var externalExtensionRegex = regexp.MustCompile(`^([a-z0-9]|[a-z0-9]_[a-z0-9])+$`)
+var internalExtensionRegex = regexp.MustCompile(`^internal:[a-zA-Z0-9]+(_[a-zA-Z0-9]+)*$`)
```

Comment:
Good catch! 

But do we want to allow uppercase characters `A-Z` too as part of this change?

Reply:
Oops :)